### ISSUE TITLE
Compact URL presentation should check actual width of input field

### ISSUE THREAD

**Issue by Altonss** [skyjake/lagrange#512]
![image](https://user-images.githubusercontent.com/66519591/171057344-fb26cc92-8247-43fc-98e3-03d2d86a9b2e.png)

**Comment by Altonss** [comment]
The bug is even stranger because when saving a page the gemini of the gemini:// but when copy pasting the "broken" URL, the gemini is then present...

**Comment by skyjake** [comment]
This is intentional. To save space, a narrow URL field will not show the default "gemini:" scheme. This applies to all URL fields in the UI.

**Comment by Altonss** [comment]
At least for me as a user this very confusing, especially because the bar is mainly empty because the url is short...

**Comment by skyjake** [comment]
I agree the implementation isn't ideal. At least it should compare the length of the actual URL to the available space and not just check the width of the field.
